src/commands/publish.rs

```diff
@@ -141,7 +165,80 @@ pub fn publish(publish_opts: PublishOpt) -> anyhow::Result<()> {
     let _compressed_archive = gz_enc.finish().unwrap();
     let mut compressed_archive_reader = fs::File::open(&archive_path)?;
 
-    let maybe_signature_data = match sign_compressed_archive(&mut compressed_archive_reader)? {
+    let maybe_signature_data = sign_compressed_archive(&mut compressed_archive_reader)?;
+    let archived_data_size = archive_path.metadata()?.len();
+    let use_chunked_uploads = archived_data_size > 1242880;
+
+    assert!(archive_path.exists());
+    assert!(archive_path.is_file());
+
+    if publish_opts.dry_run {
+        // dry run: publish is done here
+
+        println!(
+            "Successfully published package `{}@{}`",
+            package.name, package.version
+        );
+
+        info!(
+            "Publish succeeded, but package was not published because it was run in dry-run mode"
+        );
+
+        return Ok(());
+    }
+
+    // file is larger than 1MB, use chunked uploads
+    if std::env::var("FORCE_WAPM_USE_CHUNKED_UPLOAD").is_ok()
+        || (std::env::var("WAPM_USE_CHUNKED_UPLOAD").is_ok() && use_chunked_uploads)
```

Comment:
Is there any reason why we require `WAPM_USE_CHUNKED_UPLOAD` to be set instead of having chunked uploads enabled by default? 

Every package I've published to wapm.io has timed out for the last month and a half, so I'm not sure the default upload behaviour even works.

Reply:
Just to test it? I thought it could be useful to add it as an environment var while we test

Reply:
@syrusakbary I would prefer to always use chunked uploading now that we have support in wapm.io and wapm.dev.